application/src/main/java/org/togetherjava/tjbot/commands/basic/RoleSelectCommand.java

```diff
@@ -0,0 +1,320 @@
+package org.togetherjava.tjbot.commands.basic;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.MessageBuilder;
+import net.dv8tion.jda.api.Permission;
+import net.dv8tion.jda.api.entities.*;
+import net.dv8tion.jda.api.events.interaction.SelectionMenuEvent;
+import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
+import net.dv8tion.jda.api.interactions.Interaction;
+import net.dv8tion.jda.api.interactions.commands.OptionMapping;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import net.dv8tion.jda.api.interactions.commands.build.OptionData;
+import net.dv8tion.jda.api.interactions.commands.build.SubcommandData;
+import net.dv8tion.jda.api.interactions.components.ActionRow;
+import net.dv8tion.jda.api.interactions.components.ComponentInteraction;
+import net.dv8tion.jda.api.interactions.components.selections.SelectOption;
+import net.dv8tion.jda.api.interactions.components.selections.SelectionMenu;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NonNls;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.commands.SlashCommandAdapter;
+import org.togetherjava.tjbot.commands.SlashCommandVisibility;
+
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+
+
+/**
+ * Implements the {@code roleSelect} command.
+ *
+ * <p>
+ * Allows users to select their roles without using reactions, instead it uses selection menus where
+ * you can select multiple roles. <br />
+ * Note: the bot can only use roles below its highest one
+ */
+public class RoleSelectCommand extends SlashCommandAdapter {
+
+    private static final Logger logger = LoggerFactory.getLogger(RoleSelectCommand.class);
+
+    private static final String ALL_OPTION = "all";
+    private static final String CHOOSE_OPTION = "choose";
+
+    private static final String TITLE_OPTION = "title";
+    private static final String DESCRIPTION_OPTION = "description";
+
+    private static final Color embedColor = new Color(24, 221, 136);
```

Comment:
```suggestion
    private static final Color embedColor = Color.decode("#18DD88");
```

Reply:
We should use `Color#decode` so we can easily change the hex when needed and also for consistency

Reply:
I'm missing the point

Reply:
Point is we should be using hex instead of RGB, at least imo.

Reply:
And why?

Reply:
Because it is easier to edit in the future and other things too.

Reply:
What other things? And I don't see how it's easier, just use IntelliJ. 
"But what if you don't use an IDE", simple, because it's RGB you can use your brain to guess what the color is.

I doubt many people here will be able to guess what color something is by hex.


Unless better arguments come in, or people agree with you I won't change this.

Reply:
i think both are bad for humans to read and understand. everyone will either use a plugin to understand the color or just google it anyways... too much brain power needed to interpret it 😆 

that said, other commands in the code base used the hex code so far. personally, i dont mind at all and am fine with both.

a point could be made about intellij plugins showing the color and i believe intellij cant do that with the rgb but only with the hex? i might be totally wrong here though.

Reply:
Totally wrong there
![image](https://user-images.githubusercontent.com/49957334/150825576-08a105b5-dcb5-463a-84cf-f065965b926a.png)

Full support for everything
Can even change the hex in there

Reply:
> Full support for everything
Can even change the hex in there

That happens even if we use hex

Reply:
In that case I do not have any argument and officially **dont care**.

Reply:
I agree with Zabu.
If we heavily used "standard" colors, we could use a lib of sorts to get the color from an enum "ex: `IndexedColors.LIGHT_BLUE.getIndex()`", but that not being the case, both solutions have the same readability, and IDE's can easily display them

Reply:
Can this be resolved?

Reply:
I reckon so